types/web3/eth/types.d.ts

```diff
@@ -22,7 +22,7 @@ export class Iban {
 	static toAddress(iban: Iban): string;
 	isValid(): boolean;
 }
-export type BlockType = "latest" | "pending" | "genesis" | number;
+export type BlockType = "latest" | "pending" | "genesis" | number | string;
```

Comment:
This union is now equivalent to `number | string` because string is a supertype of "latest" et al.

If all users of BlockType can accept all numbers and all strings, this change is fine, but I guess (without knowing much about web3) that the reason that the string literals are there is because some uses of BlockType don't accept all possible strings.

Plus, the current union is a discriminated type, which lets you do some cool tricks. Not sure if anybody is using them though:

```ts
function h(b: number | "a" | "b"): number {
    if (b === 'a') {
        return 1
    }
    else if (b === 'b') {
        return 2
    }
    return b // works fine, b is definitely 'number'
}
```

Reply:
Just to clarify: Do you want something to be changed or do you suggest to just close this PR without further ado?
Concerning the redundant strings: I am surprised that this is not something that `tslint` complains about. Should be easy to find for a linter.

Reply:
Oops. I realise I wasn't clear. What I meant was that you need to decide between two things

1. Change BlockType to `number | string`, so it will be the same, and easier to use, everywhere.
2. Close the PR and keep the existing discriminant union: it will catch incorrect strings in locations that really only need latest/pending/genesis (but I don't know whether any locations like this exist). You also keep discriminability but this is probably not actually used anywhere.

The key point is whether some functions that use `BlockType` really care about only accepting or returning only "latest" | "pending" | "genesis". If that's true, then we should close this PR.